src/build-data/cc/clangcl.txt

```diff
@@ -0,0 +1,102 @@
+macro_name CLANGCL
+
+minimum_supported_version 14.0
+
+binary_name clang-cl
+linker_name lld-link
+
+output_to_object "/Fo"
+output_to_exe "/OUT:"
+
+add_include_dir_option "/I"
+add_system_include_dir_option "/external:W0 /external:I"
+add_lib_dir_option "/LIBPATH:"
+add_compile_definition_option "/D"
+add_lib_option "%s.lib"
+
+compile_flags "/nologo /c"
+
+supports_gcc_inline_asm yes
+
+optimization_flags "/O2 /Oi"
+size_optimization_flags "/O1 /Os"
+
+# for debug info in the object file (required if using sccache):
+#debug_info_flags "/Z7"
+
+# for using a PDB file:
+debug_info_flags "/Zi /FS"
+
+preproc_flags "/nologo /EP"
+
+lang_flags "/std:c++20 /EHs /GR"
+
+# 4251: STL types used in DLL interface
+# 4275: ???
+# 5072: ASan without debug info
+warning_flags "/W4 /wd4251 /wd4275 /wd5072"
+
+werror_flags "/WX"
+
+visibility_build_flags "/DBOTAN_DLL=__declspec(dllexport)"
+visibility_attribute "__declspec(dllimport)"
+
+# Include dependency tracking for Ninja
+# See: https://ninja-build.org/manual.html#ref_headers
+ninja_header_deps_style 'msvc'
+header_deps_flag '/showIncludes'
+
+ar_command lib
+ar_options "/nologo"
+ar_output_to "/OUT:"
+
+<sanitizers>
+default -> address
+
+iterator -> "/D_ITERATOR_DEBUG_LEVEL=1"
+address   -> "/fsanitize=address"
+</sanitizers>
+
+<isa_flags>
+sse2   -> "-msse2"
+ssse3  -> "-mssse3"
+sse41  -> "-msse4.1"
+sse42  -> "-msse4.2"
```

Comment:
Note that there's [a pull request that removes explicit (but unused) support for SSE 4.2](https://github.com/randombit/botan/pull/4331). This will likely cause issues on master. Please rebase your patch onto master once the linked PR is merged and simply remove this line.